SAPHana/test/SAPHanaSRTools.pm

```diff
@@ -653,45 +697,65 @@ sub check_all_ok($$)
 #	return 0;
 #}
 
-sub host_attr2string($$$$)
+sub host_attr2string
 {
     my $string="";
-    my ($refH, $refN, $title, $sort) = @_;
+    my ($refH, $refN, $title, $sort, $format) = @_;
     my ($len, $line_len, $hclen);
-    $hclen=$$refN{_hosts}->{_length};
-    $line_len=$hclen+1;
-    $string.=sprintf "%-$hclen.${hclen}s ", "$title";
     #
-    # headline
+    # leave function if hash is empty
+    #    in this case an empty string is returned
     #
-    foreach my $AKey (sort keys %$refN) {
-        if ($AKey ne "_hosts") {
-            $len = $$refN{$AKey}->{_length};
-            $line_len=$line_len+$len+1;
-            
-            if ( $AKey eq $sort ) {
-               $string.=sprintf "*%-$len.${len}s ", $$refN{$AKey}->{_title};
-            } else {
-               $string.=sprintf "%-$len.${len}s ", "$$refN{$AKey}->{_title}";
+    if ( ! ( keys %$refN )) {
+       return "";
+    }
+    if ( ! defined $format) {
+        $format="tables"
+    }
+    if ( $format eq "tables" ) {
+	    $hclen=$$refN{_hosts}->{_length};
+	    $line_len=$hclen+1;
+	    $string.=sprintf "%-$hclen.${hclen}s ", "$title";
+        #
+        # headline
+        #
+        foreach my $AKey (sort keys %$refN) {
+            if ($AKey ne "_hosts") {
+                $len = $$refN{$AKey}->{_length};
+                $line_len=$line_len+$len+1;
+
+                if ( $AKey eq $sort ) {
+                   $string.=sprintf "*%-$len.${len}s ", $$refN{$AKey}->{_title};
+                } else {
+                   $string.=sprintf "%-$len.${len}s ", "$$refN{$AKey}->{_title}";
+                }
             }
-        }
+	    }
+	    $string.=sprintf "\n";
+	    $string.=sprintf "%s\n", "-" x $line_len ;
     }
-    $string.=sprintf "\n";
-    $string.=sprintf "%s\n", "-" x $line_len ;
     #
     # object / name / value lines
     #
     if ( $sort eq "" ) {
         foreach my $HKey (sort keys %$refH) {
-            $string.=sprintf "%-$hclen.${hclen}s ", $HKey;
+            if ( $format eq "tables" ) {
+                $string.=sprintf "%-$hclen.${hclen}s ", $HKey;
+            }
             foreach my $AKey (sort keys %$refN) {
                 if ($AKey ne "_hosts") {
                     $len = $$refN{$AKey}->{_length};
-                    $string.=sprintf "%-$len.${len}s ", $$refH{$HKey} -> {$AKey};
+                    if ( $format eq "tables" ) {
+                        $string.=sprintf "%-$len.${len}s ", $$refH{$HKey} -> {$AKey};
+                    } elsif ( $format eq "tables" ) {
```

Comment:
Entweder hier oder in Zeile 748 ist 'tables' ein Typo. Ansonsten wäre es zweimal die gleiche Abfrage

Reply:
Das muss ich mir ansehen. Ich musste etwas am default ändern und vielleicht ist mir dabei ein c&p fehler passiert.